### ISSUE TITLE
Internet Explorer Syntax error

### ISSUE THREAD

**Issue by cdatehortuab** [simplabs/ember-simple-auth#1496]
Hello, this addon has a little issue when running a project including it in Internet Explorer.

The problem is browser says in this line: 
https://github.com/simplabs/ember-simple-auth/blob/50adfeb515f5983287695d0e3eeba814c308d79f/addon/session-stores/cookie.js#L298
there is a Syntax error.

So, the project never runs.

Thank you.

**Comment by marcoow** [comment]
This should actually be transpiled by Babel. Which version of IE do you see the error in? What does your `config/targets.js` look like?

**Comment by mdeanjones** [comment]
For a vanilla `targets.js` of

```
module.exports = {
  browsers: [
    'ie 9',
    'last 1 Chrome versions',
    'last 1 Firefox versions',
    'last 1 Safari versions'
  ]
};
```
the transpiled line becomes
```
_warn: function _warn() {
      (true && Ember.warn(...arguments));
}
```
making it the sole remaining splat in the output. Not sure if this one is on Babel.

**Comment by marcoow** [comment]
I think this relates to `@ember/debug` being a special module that has all of its exports disabled in production builds. In `development` env,

```js
_warn() {
  warn(...arguments);
}
```

becomes

```js
_warn: function _warn() {
  (true && Ember.warn(...arguments));
}
```

while in production it becomes

```js
_warn: function _warn() {
  (false && Ember.warn(...arguments));
}
```

which effectively disables the call to `Ember.warn`. Pure Babel compiles the method to

```js
_warn: function _warn() {
  _debug.warn.apply(undefined, arguments);
}
```

which seems correct. To me this looks like a bug in the AST transform doing this special handling of `@ember/debug`. @Turbo87 might have an idea whether this has maybe been fixed already or where exactly this happens.

**Comment by Turbo87** [comment]
see https://github.com/chadhietala/babel-plugin-debug-macros/issues/47

**Comment by cah-danmonroe** [comment]
Yay, I just spent the last 4 hours tracking this down to pin point it to ember-simple-auth from 1.4.0 to 1.4.1.  I came here to report it but I guess I was a little late.

IE 10 and 11 for us.

**Comment by rwjblue** [comment]
😩 sorry for not getting that upstream issue fixed already, haven't had the time to dig into it yet 😭

**Comment by andyhot** [comment]
Also getting this with phantomjs in our automated tests

**Comment by andyhot** [comment]
@marcoow Would you be open to a PR that directly calls `@ember/debug`'s `warn` instead of `this._warn` (and hopefully a quick bug release after that)?

**Comment by Turbo87** [comment]
I've opened a PR to work around the issue for now (https://github.com/simplabs/ember-simple-auth/pull/1502)

**Comment by ivp-net** [comment]
this error has killed my day (

**Comment by Turbo87** [comment]
Once CI is finished this will be released as `v1.4.2`
